members/models/address.py

```diff
@@ -121,61 +121,57 @@ def get_by_dawa_id(dawa_id):
-                for department in Department.objects.filter(
-                    adminuserinformation__user=user
-                )
-            ]
-            department_id = [
-                department.id
-                for department in Department.objects.filter(
-                    adminuserinformation__user=user
-                )
-            ]
+            department_qs = Department.objects.filter(adminuserinformation__user=user)
+        department_address_ids = set(department_qs.values_list("address_id", flat=True))
+        department_ids = set(department_qs.values_list("id", flat=True))
 
+        # Unions
         if user.has_perm("members.view_all_unions"):
-            union_address_id = [union.address.id for union in Union.objects.all()]
+            union_qs = Union.objects.all()
         else:
-            union_address_id = [
-                union.address.id
-                for union in Union.objects.filter(adminuserinformation__user=user)
-            ]
-
-        activity_address_id = []
-        for department in department_id:
-            for activity in Activity.objects.filter(department_id=department):
-                activity_address_id += [activity.address.id]
-
-        # Find all addresses not used by Union nor Department nor Activity
-        address_id_all = [address.id for address in Address.objects.all()]
-
-        department_address_id_all = [
-            department.address.id for department in Department.objects.all()
-        ]
-        union_address_id_all = [union.address.id for union in Union.objects.all()]
-        activity_address_id_all = [
-            activity.address.id for activity in Activity.objects.all()
-        ]
-
-        address_unused_ids = list(
-            set(address_id_all)
-            - set(department_address_id_all)
-            - set(union_address_id_all)
-            - set(activity_address_id_all)
+            union_qs = Union.objects.filter(adminuserinformation__user=user)
+        union_address_ids = set(union_qs.values_list("address_id", flat=True))
+
+        # Activities (only for departments user can administrate)
+        activity_address_ids = set(
+            Activity.objects.filter(department_id__in=department_ids).values_list(
+                "address_id", flat=True
+            )
+        )
+
+        # All used address IDs
+        used_address_ids = (
+            department_address_ids | union_address_ids | activity_address_ids
+        )
+
+        # Unused addresses: not linked to any department, union, or activity
+        all_address_ids = set(Address.objects.all().values_list("id", flat=True))
+        all_department_address_ids = set(
+            Department.objects.all().values_list("address_id", flat=True)
+        )
+        all_union_address_ids = set(
+            Union.objects.all().values_list("address_id", flat=True)
+        )
+        all_activity_address_ids = set(
+            Activity.objects.all().values_list("address_id", flat=True)
```

Comment:
Skal vi evt. loope henover noget af dette i stedet for at have flere forskellige linjer der gør det samme?